11-walkthrough-7.Rmd

```diff
@@ -1,209 +1,311 @@
+
+```{r dataviz tweets with positive variable}
+dv_pos <- tweets %>% 
+  # Only tweets that have the dataviz status_id
+  filter(status_id %in% unique(dv_tokens$status_id)) %>% 
+  # Is the status_id from our vector of positive word?
+  mutate(positive = if_else(status_id %in% unique(pos_tokens$status_id), 1, 0)) 
+```
 
-## Random review of tweets 
+Let's take a moment to dissect how we use `if_else` to create our `positive` column. We gave `if_else` three arguments: 
 
-```{r custom function}
-library(stringr)
+ - `status_id %in% unique(pos_tokens$status_id)`: a logical statement
+ - `1`: the value of `positive` if the logical statement is true
+ - `2` the value of `positive` if the logical statement is false 
 
-# Custom function 
- detect_pos <- function(s) {
-  sum(stringr::str_detect(s, fixed(pos_tweets$word, ignore_case = TRUE))) > 0
-}
+So our new `positive` column will take the value 1 if the `status_id` was in our `pos_tokens` dataset and the value 0 if the `status_id` was not in our `pos_tokens` dataset. Practically speaking, `positive` is 1 if the tweet has a positive word and 0 if it does not have a positive word. 
+
+And finally, let's see what percent of tweets that had "dataviz" in them also had at least one positive word: 
+
+```{r percent that had a positive word}
+dv_pos %>% 
+  count(positive) %>% 
+  mutate(perc = n / sum(n)) 
 ```
 
+About 55 percent of tweets that have "dataviz" in them also had at least one positive word and about 45 percent of them did not have at least one positive word. It's worth noting here that this finding doesn't necessarily mean users didn't have anything good to say about 45 percent of the "dataviz" tweets. We can't know precisely why some tweets had positive words and some didn't, we just know that more dataviz tweets had positive words than not. To put this in perspective, we might have a different impression if 5 percent or 95 percent of the tweets had positive words. 
+
+Since the point of exploratory data analysis is to explore and develop questions, let's continue to do that. In this last section we'll randomly select tweets to do a close reading on.
+
+### Random review of tweets 
+
+Let's review where we are so far as we work to learn more about the TidyTuesday learning community through tweets. So far we've counted frequently used words and estimated the number of tweets with positive associations. This dataset is large, so we need to zoom out and find ways to summarize the data. But it's also useful to explore by zooming in and reading some of the tweets. Reading tweets helps us to build intuition and context about how users talk about TidyTuesday in general. Even though this doesn't lead to quantitative findings, it helps us to learn more about the content we're studying and analyzing. Instead of reading all 4418 tweets, let's write some code to randomly select tweets to review. 
```

Comment:
Really, really like this point about the importance of closely reading some of the text to contextualize findings and to better understand the data